src/test/java/com/next/step/step2/StringAddCalculatorTest.java

```diff
@@ -0,0 +1,58 @@
+package com.next.step.step2;
+
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.NullSource;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class StringAddCalculatorTest {
+
+    @ParameterizedTest(name = "입력이 NULL 또는 빈 문자열인 경우")
+    @NullSource
+    @ValueSource(strings = "")
+    public void splitAndSum_null_empty_input(String input) {
+        int result = StringAddCalculator.splitAndSum(input);
+        assertThat(result).isEqualTo(0);
+    }
+
+    @ParameterizedTest(name = "입력이 숫자 하나인 경우")
+    @CsvSource(value = {"1|1", "2|2", "3|3"}, delimiter = '|')
+    public void splitAndSum_a_number_input(String input, int expected) {
+        int result = StringAddCalculator.splitAndSum(input);
+        assertThat(result).isEqualTo(expected);
+    }
+
+    @ParameterizedTest(name = "입력에 쉼표 구분자가 포함된 경우")
+    @CsvSource(value = {"1,2|3", "2,3|5", "3,4|7"}, delimiter = '|')
+    public void splitAndSum_with_comma_input(String input, int expected) {
+        int result = StringAddCalculator.splitAndSum(input);
+        assertThat(result).isEqualTo(expected);
+    }
+
+    @ParameterizedTest(name = "입력에 쉼표와 콜론 구분자가 혼합하여 포함된 경우")
+    @CsvSource(value = {"1,2:3|6", "2:3,4|9"}, delimiter = '|')
+    public void splitAndSum_with_comma_colon_input(String input, int expected) {
+        int result = StringAddCalculator.splitAndSum(input);
+        assertThat(result).isEqualTo(expected);
+    }
+
+    @Test
+    @DisplayName("입력에 커스텀 구분자가 포함된 경우")
+    public void splitAndSum_custom_delimiter_input() {
+        int result = StringAddCalculator.splitAndSum("//;\n1;2;3");
+        assertThat(result).isEqualTo(6);
+    }
+
+    @ParameterizedTest(name = "입력에 음수가 포함된 경우")
+    @ValueSource(strings = {"-1,2,3","-1:2:3","-100,-200,-300"})
```

Comment:
```suggestion
    @ValueSource(strings = {"-1,2,3", "-1:2:3", "-100,-200,-300"})
```
code format을 지켜보세요. 
IntelliJ 단축키는  Ctrl+Alt+L(윈도우), Cmd+Alt+L(맥)

---

전체를 관통하는 테스트도 좋지만, splitAndSum이 여러 기능을 하다보니, RuntimeException이 원하는 위치에서 발생했는지 놓칠 우려가 보이는데요. 관련된 메소드만 테스트하는 것은 어떨까요?
현재는 메소드가 private인데요. 이런 경우가 생길 때는 객체를 분리할 시점은 아닌가 의심해볼 필요가 있다고 생각해요.

Reply:
아 Cmd+Alt+L(맥)을 마지막에 깜빡했네요ㅠㅠ 자세히 봐주셔서 감사합니다.

RuntimeException의 범위에 대해서 생각하지 못했네요. 클래스 분리 작업을 별도로 진행하고 단위 테스트도 작성해볼게요!!